diff -r -u -w binutils-2.23.2/gas/config/tc-mips.c binutils-2.23.2-patchedmips4/gas/config/tc-mips.c
--- binutils-2.23.2/gas/config/tc-mips.c	2012-09-04 15:21:03.000000000 +0000
+++ binutils-2.23.2-patchedmips4/gas/config/tc-mips.c	2019-09-01 11:12:00.676763160 +0000
@@ -19258,7 +19258,7 @@
 
      Treat NO_ABI like the EABIs.  One reason to do this is that the
      plain 'mips' and 'mips64' configs have 'from-abi' as their default
-     architecture.  This code picks MIPS I for 'mips' and MIPS III for
+     architecture.  This code picks MIPS I for 'mips' and MIPS IV for
      'mips64', just as we did in the days before 'from-abi'.  */
   if (strcasecmp (cpu_string, "from-abi") == 0)
     {
@@ -19266,13 +19266,13 @@
 	return mips_cpu_info_from_isa (ISA_MIPS1);
 
       if (ABI_NEEDS_64BIT_REGS (mips_abi))
-	return mips_cpu_info_from_isa (ISA_MIPS3);
+	return mips_cpu_info_from_isa (ISA_MIPS4);
 
       if (file_mips_gp32 >= 0)
-	return mips_cpu_info_from_isa (file_mips_gp32 ? ISA_MIPS1 : ISA_MIPS3);
+	return mips_cpu_info_from_isa (file_mips_gp32 ? ISA_MIPS1 : ISA_MIPS4);
 
       return mips_cpu_info_from_isa (MIPS_DEFAULT_64BIT
-				     ? ISA_MIPS3
+				     ? ISA_MIPS4
 				     : ISA_MIPS1);
     }
