diff -u -r -w binutils-2.19.1/binutils/readelf.c binutils-2.19.1-patched/binutils/readelf.c
--- binutils-2.19.1/binutils/readelf.c	2008-09-17 10:00:44.000000000 +0000
+++ binutils-2.19.1-patched/binutils/readelf.c	2019-04-16 17:08:53.754992080 +0000
@@ -7886,8 +7886,10 @@
 
       if (data < end)
 	{
-#ifndef __MSVCRT__
+#if !defined(__MSVCRT__) && !defined(__sgi)
 	  printf ("  [%6tx]  %s\n", data - start, data);
+#elif defined(__sgi)
+	  printf ("  [%06x]  %s\n", (size_t) (data - start), data);
 #else
 	  printf ("  [%6Ix]  %s\n", (size_t) (data - start), data);
 #endif
