diff -u -r -w cmake-2.8.12.2/Modules/FindCurses.cmake cmake-2.8.12.2-patched/Modules/FindCurses.cmake
--- cmake-2.8.12.2/Modules/FindCurses.cmake	2014-01-16 17:15:08.000000000 +0000
+++ cmake-2.8.12.2-patched/Modules/FindCurses.cmake	2019-03-27 00:00:35.800165332 +0000
@@ -11,6 +11,7 @@
 #
 # Set CURSES_NEED_NCURSES to TRUE before the find_package() command if NCurses
 # functionality is required.
+set(CURSES_NEED_NCURSES TRUE)
 
 #=============================================================================
 # Copyright 2001-2009 Kitware, Inc.
@@ -28,7 +29,7 @@
 find_library(CURSES_CURSES_LIBRARY NAMES curses )
 
 find_library(CURSES_NCURSES_LIBRARY NAMES ncurses )
-set(CURSES_USE_NCURSES FALSE)
+set(CURSES_USE_NCURSES TRUE)
 
 if(CURSES_NCURSES_LIBRARY  AND NOT  CURSES_CURSES_LIBRARY)
   set(CURSES_USE_NCURSES TRUE)
@@ -94,10 +95,7 @@
   find_file(CURSES_HAVE_CURSES_H          curses.h
     HINTS "${_cursesParentDir}/include")
 
-  find_path(CURSES_NCURSES_INCLUDE_PATH ncurses.h ncurses/ncurses.h
-    ncurses/curses.h)
-  find_path(CURSES_NCURSES_INCLUDE_PATH curses.h
-    HINTS "${_cursesParentDir}/include")
+  set(CURSES_NCURSES_INCLUDE_PATH "${_cursesParentDir}/include")
 
   # for compatibility with older FindCurses.cmake this has to be in the cache
   # FORCE must not be used since this would break builds which preload
diff -u -r -w cmake-2.8.12.2/Source/CursesDialog/CMakeLists.txt cmake-2.8.12.2-patched/Source/CursesDialog/CMakeLists.txt
--- cmake-2.8.12.2/Source/CursesDialog/CMakeLists.txt	2014-01-16 17:15:08.000000000 +0000
+++ cmake-2.8.12.2-patched/Source/CursesDialog/CMakeLists.txt	2019-03-27 00:00:35.800165332 +0000
@@ -27,7 +27,7 @@
 
 include_directories(${CMake_SOURCE_DIR}/Source/CursesDialog/form
                     ${CMake_BINARY_DIR}/Source/CursesDialog/form)
-include_directories(${CURSES_INCLUDE_PATH})
+include_directories(${CURSES_NCURSES_INCLUDE_PATH})
 
 
 add_executable(ccmake ${CURSES_SRCS} )
diff -u -r -w cmake-2.8.12.2/Source/CursesDialog/form/CMakeLists.txt cmake-2.8.12.2-patched/Source/CursesDialog/form/CMakeLists.txt
--- cmake-2.8.12.2/Source/CursesDialog/form/CMakeLists.txt	2014-01-16 17:15:08.000000000 +0000
+++ cmake-2.8.12.2-patched/Source/CursesDialog/form/CMakeLists.txt	2019-03-27 00:00:35.800165332 +0000
@@ -12,7 +12,7 @@
 project(CMAKE_FORM)
 
 include_regular_expression("^.*$")
-include_directories(${CURSES_INCLUDE_PATH} "${CMAKE_CURRENT_BINARY_DIR}")
+include_directories(${CURSES_NCURSES_INCLUDE_PATH} "${CMAKE_CURRENT_BINARY_DIR}")
 
 configure_file(cmFormConfigure.h.in "${CMAKE_CURRENT_BINARY_DIR}/cmFormConfigure.h")
 
diff -u -r -w cmake-2.8.12.2/Tests/CMakeLists.txt cmake-2.8.12.2-patched/Tests/CMakeLists.txt
--- cmake-2.8.12.2/Tests/CMakeLists.txt	2014-01-16 17:15:08.000000000 +0000
+++ cmake-2.8.12.2-patched/Tests/CMakeLists.txt	2019-03-27 00:00:53.087428023 +0000
@@ -262,9 +262,9 @@
 
   list(APPEND TEST_BUILD_DIRS ${CMake_TEST_INSTALL_PREFIX})
 
-  if(NOT QT4_FOUND)
-    find_package(Qt4)
-  endif()
+#  if(NOT QT4_FOUND)
+#    find_package(Qt4)
+#  endif()
 
   if(QT4_FOUND)
     # test whether the Qt4 which has been found works, on some machines
@@ -1003,72 +1003,72 @@
     )
   list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/Environment")
 
-  add_test(QtAutomocNoQt  ${CMAKE_CTEST_COMMAND}
-    --build-and-test
-    "${CMake_SOURCE_DIR}/Tests/QtAutomocNoQt"
-    "${CMake_BINARY_DIR}/Tests/QtAutomocNoQt"
-    ${build_generator_args}
-    --build-project QtAutomocNoQt
-    --build-options -DCMAKE_BUILD_TYPE=\${CTEST_CONFIGURATION_TYPE}
-    )
-  list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/QtAutomocNoQt")
+#  add_test(QtAutomocNoQt  ${CMAKE_CTEST_COMMAND}
+#    --build-and-test
+#    "${CMake_SOURCE_DIR}/Tests/QtAutomocNoQt"
+#    "${CMake_BINARY_DIR}/Tests/QtAutomocNoQt"
+#    ${build_generator_args}
+#    --build-project QtAutomocNoQt
+#    --build-options -DCMAKE_BUILD_TYPE=\${CTEST_CONFIGURATION_TYPE}
+#    )
+#  list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/QtAutomocNoQt")
 
-  find_package(Qt5Widgets QUIET NO_MODULE)
-  if(Qt5Widgets_FOUND)
-    add_test(Qt5Automoc ${CMAKE_CTEST_COMMAND}
-      --build-and-test
-      "${CMake_SOURCE_DIR}/Tests/QtAutomoc"
-      "${CMake_BINARY_DIR}/Tests/Qt5Automoc"
-      ${build_generator_args}
-      --build-project QtAutomoc
-      --build-exe-dir "${CMake_BINARY_DIR}/Tests/Qt5Automoc"
-      --force-new-ctest-process
-      --build-options -DQT_QMAKE_EXECUTABLE:FILEPATH=${QT_QMAKE_EXECUTABLE} -DQT_TEST_VERSION=5
-      --test-command ${CMAKE_CTEST_COMMAND} -V
-      )
-    list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/Qt5Automoc")
-  endif()
-  if(QT4_WORKS AND QT_QTGUI_FOUND)
-    add_test(Qt4Automoc ${CMAKE_CTEST_COMMAND}
-      --build-and-test
-      "${CMake_SOURCE_DIR}/Tests/QtAutomoc"
-      "${CMake_BINARY_DIR}/Tests/Qt4Automoc"
-      ${build_generator_args}
-      --build-project QtAutomoc
-      --build-exe-dir "${CMake_BINARY_DIR}/Tests/Qt4Automoc"
-      --force-new-ctest-process
-      --build-options -DQT_QMAKE_EXECUTABLE:FILEPATH=${QT_QMAKE_EXECUTABLE} -DQT_TEST_VERSION=4
-      --test-command ${CMAKE_CTEST_COMMAND} -V
-      )
-    list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/Qt4Automoc")
+#  find_package(Qt5Widgets QUIET NO_MODULE)
+#  if(Qt5Widgets_FOUND)
+#    add_test(Qt5Automoc ${CMAKE_CTEST_COMMAND}
+#      --build-and-test
+#      "${CMake_SOURCE_DIR}/Tests/QtAutomoc"
+#      "${CMake_BINARY_DIR}/Tests/Qt5Automoc"
+#      ${build_generator_args}
+#      --build-project QtAutomoc
+#      --build-exe-dir "${CMake_BINARY_DIR}/Tests/Qt5Automoc"
+#      --force-new-ctest-process
+#      --build-options -DQT_QMAKE_EXECUTABLE:FILEPATH=${QT_QMAKE_EXECUTABLE} -DQT_TEST_VERSION=5
+#      --test-command ${CMAKE_CTEST_COMMAND} -V
+#      )
+#    list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/Qt5Automoc")
+#  endif()
+#  if(QT4_WORKS AND QT_QTGUI_FOUND)
+#    add_test(Qt4Automoc ${CMAKE_CTEST_COMMAND}
+#      --build-and-test
+#      "${CMake_SOURCE_DIR}/Tests/QtAutomoc"
+#      "${CMake_BINARY_DIR}/Tests/Qt4Automoc"
+#      ${build_generator_args}
+#      --build-project QtAutomoc
+#      --build-exe-dir "${CMake_BINARY_DIR}/Tests/Qt4Automoc"
+#      --force-new-ctest-process
+#      --build-options -DQT_QMAKE_EXECUTABLE:FILEPATH=${QT_QMAKE_EXECUTABLE} -DQT_TEST_VERSION=4
+#      --test-command ${CMAKE_CTEST_COMMAND} -V
+#      )
+#    list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/Qt4Automoc")
 
-    add_test(Qt4Targets ${CMAKE_CTEST_COMMAND}
-      --build-and-test
-      "${CMake_SOURCE_DIR}/Tests/Qt4Targets"
-      "${CMake_BINARY_DIR}/Tests/Qt4Targets"
-      ${build_generator_args}
-      --build-project Qt4Targets
-      --build-exe-dir "${CMake_BINARY_DIR}/Tests/Qt4Targets"
-      --force-new-ctest-process
-      --build-options -DQT_QMAKE_EXECUTABLE:FILEPATH=${QT_QMAKE_EXECUTABLE}
-      --test-command ${CMAKE_CTEST_COMMAND} -V
-      )
-    list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/Qt4Targets")
+#    add_test(Qt4Targets ${CMAKE_CTEST_COMMAND}
+#      --build-and-test
+#      "${CMake_SOURCE_DIR}/Tests/Qt4Targets"
+#      "${CMake_BINARY_DIR}/Tests/Qt4Targets"
+#      ${build_generator_args}
+#      --build-project Qt4Targets
+#      --build-exe-dir "${CMake_BINARY_DIR}/Tests/Qt4Targets"
+#      --force-new-ctest-process
+#      --build-options -DQT_QMAKE_EXECUTABLE:FILEPATH=${QT_QMAKE_EXECUTABLE}
+#      --test-command ${CMAKE_CTEST_COMMAND} -V
+#      )
+#    list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/Qt4Targets")
 
-    if(Qt5Widgets_FOUND AND NOT Qt5Widgets_VERSION VERSION_LESS 5.1.0)
-      add_test(Qt4And5Automoc ${CMAKE_CTEST_COMMAND}
-        --build-and-test
-        "${CMake_SOURCE_DIR}/Tests/Qt4And5Automoc"
-        "${CMake_BINARY_DIR}/Tests/Qt4And5Automoc"
-        ${build_generator_args}
-        --build-project Qt4And5Automoc
-        --build-exe-dir "${CMake_BINARY_DIR}/Tests/Qt4And5Automoc"
-        --force-new-ctest-process
-        --test-command ${CMAKE_CTEST_COMMAND} -V
-        )
-      list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/Qt4And5Automoc")
-    endif()
-  endif()
+#    if(Qt5Widgets_FOUND AND NOT Qt5Widgets_VERSION VERSION_LESS 5.1.0)
+#      add_test(Qt4And5Automoc ${CMAKE_CTEST_COMMAND}
+#        --build-and-test
+#        "${CMake_SOURCE_DIR}/Tests/Qt4And5Automoc"
+#        "${CMake_BINARY_DIR}/Tests/Qt4And5Automoc"
+#        ${build_generator_args}
+#        --build-project Qt4And5Automoc
+#        --build-exe-dir "${CMake_BINARY_DIR}/Tests/Qt4And5Automoc"
+#        --force-new-ctest-process
+#        --test-command ${CMAKE_CTEST_COMMAND} -V
+#        )
+#      list(APPEND TEST_BUILD_DIRS "${CMake_BINARY_DIR}/Tests/Qt4And5Automoc")
+#    endif()
+#  endif()
 
   add_test(ExternalProject ${CMAKE_CTEST_COMMAND}
     --build-and-test
diff -u -r -w cmake-2.8.12.2/Tests/RunCMake/CMakeLists.txt cmake-2.8.12.2-patched/Tests/RunCMake/CMakeLists.txt
--- cmake-2.8.12.2/Tests/RunCMake/CMakeLists.txt	2014-01-16 17:15:08.000000000 +0000
+++ cmake-2.8.12.2-patched/Tests/RunCMake/CMakeLists.txt	2019-03-27 00:00:35.803498780 +0000
@@ -101,15 +101,15 @@
 add_RunCMake_test(TargetPolicies)
 add_RunCMake_test(alias_targets)
 
-find_package(Qt4 QUIET)
-find_package(Qt5Core QUIET)
-if (QT4_FOUND AND Qt5Core_FOUND AND NOT Qt5Core_VERSION VERSION_LESS 5.1.0)
-  add_RunCMake_test(IncompatibleQt)
-endif()
-if (QT4_FOUND)
-  set(ObsoleteQtMacros_ARGS -DQT_QMAKE_EXECUTABLE:FILEPATH=${QT_QMAKE_EXECUTABLE})
-  add_RunCMake_test(ObsoleteQtMacros)
-endif()
+#find_package(Qt4 QUIET)
+#find_package(Qt5Core QUIET)
+#if (QT4_FOUND AND Qt5Core_FOUND AND NOT Qt5Core_VERSION VERSION_LESS 5.1.0)
+#  add_RunCMake_test(IncompatibleQt)
+#endif()
+#if (QT4_FOUND)
+#  set(ObsoleteQtMacros_ARGS -DQT_QMAKE_EXECUTABLE:FILEPATH=${QT_QMAKE_EXECUTABLE})
+#  add_RunCMake_test(ObsoleteQtMacros)
+#endif()
 
 if("${CMAKE_TEST_GENERATOR}" MATCHES "Visual Studio [^6]")
   add_RunCMake_test(include_external_msproject)
