diff -r -u -w wget-1.20.3/Makefile.am wget-1.20.3-patched/Makefile.am
--- wget-1.20.3/Makefile.am	2019-02-10 10:48:02.000000000 +0000
+++ wget-1.20.3-patched/Makefile.am	2019-11-02 14:28:37.532582960 +0000
@@ -40,7 +40,7 @@
 ACLOCAL_AMFLAGS = -I m4
 
 # subdirectories in the distribution
-SUBDIRS = lib src doc po util fuzz tests testenv
+SUBDIRS = lib src doc po util tests testenv
 
 EXTRA_DIST = MAILING-LIST \
              msdos/config.h msdos/Makefile.DJ \
diff -r -u -w wget-1.20.3/Makefile.in wget-1.20.3-patched/Makefile.in
--- wget-1.20.3/Makefile.in	2019-04-05 11:01:26.000000000 +0000
+++ wget-1.20.3-patched/Makefile.in	2019-11-02 14:28:51.572670400 +0000
@@ -1469,7 +1469,7 @@
 ACLOCAL_AMFLAGS = -I m4
 
 # subdirectories in the distribution
-SUBDIRS = lib src doc po util fuzz tests testenv
+SUBDIRS = lib src doc po util tests testenv
 EXTRA_DIST = MAILING-LIST \
              msdos/config.h msdos/Makefile.DJ \
              msdos/Makefile.WC ABOUT-NLS \
