diff -u -r -w wget-1.20.1/Makefile.am wget-1.20.1-patched/Makefile.am
--- wget-1.20.1/Makefile.am	2018-12-26 19:58:38.000000000 +0000
+++ wget-1.20.1-patched/Makefile.am	2019-02-04 12:44:18.526896040 +0000
@@ -41,7 +41,7 @@
 ACLOCAL_AMFLAGS = -I m4
 
 # subdirectories in the distribution
-SUBDIRS = lib src doc po util fuzz tests testenv
+SUBDIRS = lib src doc po util tests testenv
 
 EXTRA_DIST = MAILING-LIST \
              msdos/config.h msdos/Makefile.DJ \
diff -u -r -w wget-1.20.1/Makefile.in wget-1.20.1-patched/Makefile.in
--- wget-1.20.1/Makefile.in	2018-12-26 20:09:02.000000000 +0000
+++ wget-1.20.1-patched/Makefile.in	2019-02-04 12:44:29.481204400 +0000
@@ -1460,7 +1460,7 @@
 ACLOCAL_AMFLAGS = -I m4
 
 # subdirectories in the distribution
-SUBDIRS = lib src doc po util fuzz tests testenv
+SUBDIRS = lib src doc po util tests testenv
 EXTRA_DIST = MAILING-LIST \
              msdos/config.h msdos/Makefile.DJ \
              msdos/Makefile.WC ABOUT-NLS \
diff -u -r -w wget-1.20.1/fuzz/Makefile.am wget-1.20.1-patched/fuzz/Makefile.am
--- wget-1.20.1/fuzz/Makefile.am	2018-12-26 19:58:38.000000000 +0000
+++ wget-1.20.1-patched/fuzz/Makefile.am	2019-02-04 12:43:37.331315760 +0000
@@ -1,4 +1,4 @@
-AM_CFLAGS = $(WERROR_CFLAGS) $(WARN_CFLAGS) -Wno-unused-parameter -Wno-pedantic
+AM_CFLAGS = $(WERROR_CFLAGS) $(WARN_CFLAGS)
 AM_CPPFLAGS = -I$(top_srcdir)/src -I$(srcdir) -I$(top_builddir)/lib -I$(top_srcdir)/lib \
  -DSRCDIR=\"$(abs_srcdir)\"
 LDADD = ../lib/libgnu.a \
diff -u -r -w wget-1.20.1/fuzz/Makefile.in wget-1.20.1-patched/fuzz/Makefile.in
--- wget-1.20.1/fuzz/Makefile.in	2018-12-26 20:09:02.000000000 +0000
+++ wget-1.20.1-patched/fuzz/Makefile.in	2019-02-04 12:43:37.340855920 +0000
@@ -1656,7 +1656,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CFLAGS = $(WERROR_CFLAGS) $(WARN_CFLAGS) -Wno-unused-parameter -Wno-pedantic
+AM_CFLAGS = $(WERROR_CFLAGS) $(WARN_CFLAGS)
 AM_CPPFLAGS = -I$(top_srcdir)/src -I$(srcdir) -I$(top_builddir)/lib \
 	-I$(top_srcdir)/lib -DSRCDIR=\"$(abs_srcdir)\" $(am__append_2)
 LDADD = ../lib/libgnu.a $(GETADDRINFO_LIB) $(HOSTENT_LIB) \
