diff -u -r -w binutils-2.23.2/bfd/cache.c binutils-2.23.2-patched/bfd/cache.c
--- binutils-2.23.2/bfd/cache.c	2013-03-25 08:06:19.000000000 +0000
+++ binutils-2.23.2-patched/bfd/cache.c	2019-01-27 00:21:24.237575920 +0000
@@ -437,7 +437,10 @@
         {
           *map_addr = ret;
           *map_len = pg_len;
-          ret += offset & pagesize_m1;
+	  uint64_t tmpVar = (offset & pagesize_m1);
+	  uint8_t * retTmpPtr = (uint8_t*)ret;
+	  retTmpPtr = retTmpPtr + tmpVar;
+          ret = (void*)(retTmpPtr);
         }
     }
 #endif
diff -u -r -w binutils-2.23.2/binutils/readelf.c binutils-2.23.2-patched/binutils/readelf.c
--- binutils-2.23.2/binutils/readelf.c	2013-03-25 08:06:21.000000000 +0000
+++ binutils-2.23.2-patched/binutils/readelf.c	2019-01-27 00:21:41.016470800 +0000
@@ -7286,7 +7286,10 @@
 
   for (i = 0; handlers[i].handler != NULL; i++)
     if (elf_header.e_machine == handlers[i].machtype)
-      return handlers[i].handler (file);
+      {
+	handlers[i].handler (file);
+	return;
+      }
 
   printf (_("\nThe decoding of unwind sections for machine type %s is not currently supported.\n"),
 	  get_machine_name (elf_header.e_machine));
diff -u -r -w binutils-2.23.2/config/override.m4 binutils-2.23.2-patched/config/override.m4
--- binutils-2.23.2/config/override.m4	2011-03-05 07:33:13.000000000 +0000
+++ binutils-2.23.2-patched/config/override.m4	2019-01-27 00:21:24.239715760 +0000
@@ -29,7 +29,7 @@
 
 dnl Ensure exactly this Autoconf version is used
 m4_ifndef([_GCC_AUTOCONF_VERSION],
-  [m4_define([_GCC_AUTOCONF_VERSION], [2.64])])
+  [m4_define([_GCC_AUTOCONF_VERSION], [2.69])])
 
 dnl Test for the exact version when AC_INIT is expanded.
 dnl This allows to update the tree in steps (for testing)
