diff -u -r -w gnutls-3.5.19/lib/priority.c gnutls-3.5.19-patched/lib/priority.c
--- gnutls-3.5.19/lib/priority.c	2018-07-16 07:13:09.000000000 +0000
+++ gnutls-3.5.19-patched/lib/priority.c	2019-03-14 18:15:57.882374640 +0000
@@ -1057,7 +1057,7 @@
 			} while (l>0);
 
 			_gnutls_debug_log("resolved '%.*s' to '%s', next '%.*s'\n",
-					  ss_len, ss, p ? : "", ss_next_len, ss_next ? : "");
+					  ss_len, ss, p ? p : "", ss_next_len, ss_next ? ss_next : "");
 			ss = ss_next;
 			fclose(fp);
 			fp = NULL;
diff -u -r -w gnutls-3.5.19/tests/utils.c gnutls-3.5.19-patched/tests/utils.c
--- gnutls-3.5.19/tests/utils.c	2017-04-05 19:07:41.000000000 +0000
+++ gnutls-3.5.19-patched/tests/utils.c	2019-03-14 18:15:41.567500560 +0000
@@ -92,6 +92,41 @@
     "aXgH9staP0HXCyyW1VAyqo0TwcWDhE/R7IQQGGwGyd4rD0T+ySW/t09ox23O6X8J\n"
     "FSp6mOVNcuvhB5U2gW8CAgEA\n" "-----END DH PARAMETERS-----\n";
 
+void _vfail(const char *format, va_list arg_ptr)
+{
+#ifdef HAVE_VASPRINTF
+	char *str = NULL;
+	vasprintf(&str, format, arg_ptr);
+
+	if (str)
+		fputs(str, stderr);
+#else
+	{
+		char str[1024];
+
+		vsnprintf(str, sizeof(str), format, arg_ptr);
+		fputs(str, stderr);
+	}
+#endif
+	va_end(arg_ptr);
+	error_count++;
+	exit(1);
+}
+
+void fail(const char *format, ...)
+{
+	va_list arg_ptr;
+	va_start(arg_ptr, format);
+	_vfail(format, arg_ptr);
+}
+
+void test_fail(const char *format, ...)
+{
+	va_list arg_ptr;
+	va_start(arg_ptr, format);
+	_vfail(format, arg_ptr);
+}
+
 void _fail(const char *format, ...)
 {
 	va_list arg_ptr;
@@ -160,6 +195,21 @@
 	fputs(str, stderr);
 }
 
+void _vsuccess(const char *format, va_list arg_ptr)
+{
+	char str[1024];
+	vsnprintf(str, sizeof(str), format, arg_ptr);
+	va_end(arg_ptr);
+	fputs(str, stderr);
+}
+
+void test_success(const char *format, ...)
+{
+	va_list arg_ptr;
+	va_start(arg_ptr, format);
+	_vsuccess(format, arg_ptr);
+}
+
 void escapeprint(const char *str, size_t len)
 {
 	size_t i;
diff -u -r -w gnutls-3.5.19/tests/utils.h gnutls-3.5.19-patched/tests/utils.h
--- gnutls-3.5.19/tests/utils.h	2016-12-19 14:10:59.000000000 +0000
+++ gnutls-3.5.19-patched/tests/utils.h	2019-03-14 18:15:30.061351440 +0000
@@ -53,22 +53,30 @@
 extern const char *pkcs3_2048;
 extern const char *pkcs3_3072;
 
+/*
 #define fail(format, ...) \
     _fail("%s:%d: "format, __func__, __LINE__, ##__VA_ARGS__)
+*/
+extern void fail(const char *format, ...);
+extern void test_fail(const char *format, ...);
+
 
 extern void _fail(const char *format, ...)
     __attribute__ ((format(printf, 1, 2)));
 extern void fail_ignore(const char *format, ...)
     __attribute__ ((format(printf, 1, 2)));
-extern void success(const char *format, ...)
-    __attribute__ ((format(printf, 1, 2)));
+extern void success(const char *format, ...);
+//    __attribute__ ((format(printf, 1, 2)));
 
 /* assumes test_name is defined */
+/*
 #define test_fail(fmt, ...) \
 	fail("%s: "fmt, test_name, ##__VA_ARGS__)
 
+
 #define test_success(fmt, ...) \
 	success("%s: "fmt, test_name, ##__VA_ARGS__)
+*/
 
 extern void c_print(const unsigned char *str, size_t len);
 extern void escapeprint(const char *str, size_t len);
