diff -r -u -w ntp-4.2.8p11/include/ntp_request.h ntp-4.2.8p11-patched/include/ntp_request.h
--- ntp-4.2.8p11/include/ntp_request.h	2018-02-27 15:15:24.000000000 +0000
+++ ntp-4.2.8p11-patched/include/ntp_request.h	2018-08-11 01:21:41.523556440 +0000
@@ -662,6 +662,9 @@
 	struct in6_addr mask6; 	/* match mask (v6) */
 };
 
+#if defined(__sgi)
+#pragma pack 4
+#endif
 
 /*
  * Structure used for specifying restrict entries
@@ -677,6 +680,9 @@
 	struct in6_addr mask6; 	/* match mask (v6) */
 };
 
+#if defined(__sgi)
+#pragma pack 0
+#endif
 
 /*
  * Structure used for returning monitor data
Only in ntp-4.2.8p11-patched/include: ntp_request.h.orig
diff -r -u -w ntp-4.2.8p11/tests/libntp/a_md5encrypt.c ntp-4.2.8p11-patched/tests/libntp/a_md5encrypt.c
--- ntp-4.2.8p11/tests/libntp/a_md5encrypt.c	2018-02-27 15:15:45.000000000 +0000
+++ ntp-4.2.8p11-patched/tests/libntp/a_md5encrypt.c	2018-08-11 01:21:54.137336840 +0000
@@ -89,12 +89,22 @@
 void
 test_IPv6AddressToRefId(void) {
 	const int expected = 0x75cffd52;
+#if !defined(__sgi)
 	const struct in6_addr address = { { {
 		0x20, 0x01, 0x0d, 0xb8,
 		0x85, 0xa3, 0x08, 0xd3,
 		0x13, 0x19, 0x8a, 0x2e,
 		0x03, 0x70, 0x73, 0x34
 	} } };
+#else
+	const struct in6_addr address;
+	address.u6_addr8 = {
+		0x20, 0x01, 0x0d, 0xb8,
+		0x85, 0xa3, 0x08, 0xd3,
+		0x13, 0x19, 0x8a, 0x2e,
+		0x03, 0x70, 0x73, 0x34
+	};
+#endif
 	sockaddr_u addr;
 
 	addr.sa6.sin6_family = AF_INET6;
