diff -u -r -w gdb-6.3/gdb/defs.h gdb-6.3-patched/gdb/defs.h
--- gdb-6.3/gdb/defs.h	2004-10-12 11:06:14.000000000 +0000
+++ gdb-6.3-patched/gdb/defs.h	2019-04-06 06:47:01.400205640 +0000
@@ -50,6 +50,12 @@
 /* For ``enum target_signal''.  */
 #include "gdb/signals.h"
 
+#include <sys/signal.h>
+
+#if !defined(NSIG)
+#define NSIG SIGRTMAX
+#endif
+
 /* Just in case they're not defined in stdio.h.  */
 
 #ifndef SEEK_SET
Only in gdb-6.3-patched/gdb: defs.h.orig
diff -u -r -w gdb-6.3/gdb/irix5-nat.c gdb-6.3-patched/gdb/irix5-nat.c
--- gdb-6.3/gdb/irix5-nat.c	2004-08-03 03:02:22.000000000 +0000
+++ gdb-6.3-patched/gdb/irix5-nat.c	2019-04-06 07:22:34.029208600 +0000
@@ -36,6 +36,8 @@
 #include <sys/procfs.h>
 #include <setjmp.h>		/* For JB_XXX.  */
 
+#define	JB_PC		1	/* program counter */
+
 /* Prototypes for supply_gregset etc. */
 #include "gregset.h"
 #include "mips-tdep.h"
Only in gdb-6.3-patched/gdb: irix5-nat.c~
diff -u -r -w gdb-6.3/libiberty/strsignal.c gdb-6.3-patched/libiberty/strsignal.c
--- gdb-6.3/libiberty/strsignal.c	2003-04-15 22:29:34.000000000 +0000
+++ gdb-6.3-patched/libiberty/strsignal.c	2019-04-06 06:47:01.404035240 +0000
@@ -253,6 +253,8 @@
 #else
 #ifdef _NSIG
 static int sys_nsig = _NSIG;
+#else
+static int sys_nsig = SIGRTMAX;
 #endif
 #endif
 extern const char * const sys_siglist[];
Only in gdb-6.3-patched/libiberty: strsignal.c.orig
diff -u -r -w gdb-6.3/sim/common/sim-arange.c gdb-6.3-patched/sim/common/sim-arange.c
--- gdb-6.3/sim/common/sim-arange.c	1999-12-14 01:06:04.000000000 +0000
+++ gdb-6.3-patched/sim/common/sim-arange.c	2019-04-06 06:47:01.406621000 +0000
@@ -281,11 +281,7 @@
   build_search_tree (ar);
 }
 
-#endif /* DEFINE_NON_INLINE_P */
-
-#if DEFINE_INLINE_P
-
-SIM_ARANGE_INLINE int
+int
 sim_addr_range_hit_p (ADDR_RANGE *ar, address_word addr)
 {
   ADDR_RANGE_TREE *t = ar->range_tree;
@@ -302,4 +298,4 @@
   return 0;
 }
 
-#endif /* DEFINE_INLINE_P */
+#endif /* DEFINE_NON_INLINE_P */
Only in gdb-6.3-patched/sim/common: sim-arange.c.orig
diff -u -r -w gdb-6.3/sim/common/sim-arange.h gdb-6.3-patched/sim/common/sim-arange.h
--- gdb-6.3/sim/common/sim-arange.h	1999-04-16 02:34:57.000000000 +0000
+++ gdb-6.3-patched/sim/common/sim-arange.h	2019-04-06 06:47:01.408756120 +0000
@@ -63,7 +63,7 @@
 
 /* Return non-zero if ADDR is in range AR, traversing the entire tree.
    If no range is specified, that is defined to mean "everything".  */
-extern INLINE int
+extern int
 sim_addr_range_hit_p (ADDR_RANGE * /*ar*/, address_word /*addr*/);
 #define ADDR_RANGE_HIT_P(ar, addr) \
   ((ar)->range_tree == NULL || sim_addr_range_hit_p ((ar), (addr)))
Only in gdb-6.3-patched/sim/common: sim-arange.h.orig
