diff -N -r -u -w fakechroot-2.20.1/scripts/env.fakechroot.sh fakechroot-2.20.1-patched/scripts/env.fakechroot.sh
--- fakechroot-2.20.1/scripts/env.fakechroot.sh	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/scripts/env.fakechroot.sh	2019-10-04 06:57:52.029171120 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 # env
 #
diff -N -r -u -w fakechroot-2.20.1/scripts/fakechroot.sh fakechroot-2.20.1-patched/scripts/fakechroot.sh
--- fakechroot-2.20.1/scripts/fakechroot.sh	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/scripts/fakechroot.sh	2019-10-04 06:58:13.428312000 +0000
@@ -1,4 +1,4 @@
-#!@SHELL@
+#!/usr/bin/env bash
 
 # fakechroot
 #
@@ -87,7 +87,8 @@
     fakechroot_echo=`which echo`
     fakechroot_echo=${fakechroot_echo:-@ECHO@}
 else
-    fakechroot_echo=@ECHO@
+#    fakechroot_echo=@ECHO@
+    fakechroot_echo=/usr/didbs/0_1_5_n32_mips4_mp/bin/echo
 fi
 
 
@@ -190,10 +191,14 @@
 
 
 # Make sure the preload is available
-fakechroot_paths="$fakechroot_paths${LD_LIBRARY_PATH:+${fakechroot_paths:+:}$LD_LIBRARY_PATH}"
+#fakechroot_paths="$fakechroot_paths${LD_LIBRARY_PATH:+${fakechroot_paths:+:}$LD_LIBRARY_PATH}"
+fakechroot_paths="$fakechroot_paths${LD_LIBRARYN32_PATH:+${fakechroot_paths:+:}$LD_LIBRARYN32_PATH}"
 fakechroot_lib="${lib_libfakeroot_to_preload:+${lib_libfakeroot_to_preload}:}$fakechroot_lib${lib_to_preload:+:$lib_to_preload}"
 
-fakechroot_detect=`LD_LIBRARY_PATH="$fakechroot_paths" LD_PRELOAD="$fakechroot_lib" FAKECHROOT_DETECT=1 $fakechroot_echo 2>&1`
+#fakechroot_detect=`LD_LIBRARY_PATH="$fakechroot_paths" LD_PRELOAD="$fakechroot_lib" FAKECHROOT_DETECT=1 $fakechroot_echo 2>&1`
+thestring="_RLDN32_ROOT=\"/\" LD_LIBRARYN32_PATH=\"$fakechroot_paths\" _RLDN32_LIST=\"$fakechroot_lib:DEFAULT\" FAKECHROOT_DETECT=1 $fakechroot_echo 2>&1"
+fakechroot_detect=`_RLDN32_ROOT="/" LD_LIBRARYN32_PATH="$fakechroot_paths" _RLDN32_LIST="$fakechroot_lib:DEFAULT" FAKECHROOT_DETECT=1 $fakechroot_echo 2>&1`
+$fakechroot_echo "DH $thestring DH"
 case "$fakechroot_detect" in
     fakechroot*)
         fakechroot_libfound=yes
diff -N -r -u -w fakechroot-2.20.1/scripts/restoremode.sh fakechroot-2.20.1-patched/scripts/restoremode.sh
--- fakechroot-2.20.1/scripts/restoremode.sh	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/scripts/restoremode.sh	2019-10-04 06:57:52.037208880 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 # This script restores uids and gids of files saved previously
 # with savemode.sh script
diff -N -r -u -w fakechroot-2.20.1/src/bind.c fakechroot-2.20.1-patched/src/bind.c
--- fakechroot-2.20.1/src/bind.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/bind.c	2019-10-04 06:57:52.041308800 +0000
@@ -42,7 +42,11 @@
 #endif
 
 
+#if defined(__sgi)
+wrapper(bind, int, (int sockfd, const void *addr, socklen_t addrlen))
+#else
 wrapper(bind, int, (int sockfd, BIND_TYPE_ARG2(addr), socklen_t addrlen))
+#endif
 {
     char fakechroot_abspath[FAKECHROOT_PATH_MAX];
     char fakechroot_buf[FAKECHROOT_PATH_MAX];
diff -N -r -u -w fakechroot-2.20.1/src/connect.c fakechroot-2.20.1-patched/src/connect.c
--- fakechroot-2.20.1/src/connect.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/connect.c	2019-10-04 06:57:52.052522400 +0000
@@ -42,7 +42,11 @@
 #endif
 
 
+#if defined(__sgi)
+wrapper(connect, int, (int sockfd, const void *addr, socklen_t addrlen))
+#else
 wrapper(connect, int, (int sockfd, CONNECT_TYPE_ARG2(addr), socklen_t addrlen))
+#endif
 {
     char fakechroot_abspath[FAKECHROOT_PATH_MAX];
     char fakechroot_buf[FAKECHROOT_PATH_MAX];
diff -N -r -u -w fakechroot-2.20.1/src/fts.c fakechroot-2.20.1-patched/src/fts.c
--- fakechroot-2.20.1/src/fts.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/fts.c	2019-10-04 06:57:52.059760320 +0000
@@ -32,7 +32,7 @@
 
 #include <config.h>
 
-#if !defined FTS64_C__ || (defined FTS64_C__ && HAVE_FTS64_OPEN)
+#if !defined(__sgi) && (!defined FTS64_C__ || (defined FTS64_C__ && HAVE_FTS64_OPEN))
 
 #define _ATFILE_SOURCE
 #define _BSD_SOURCE
@@ -45,7 +45,9 @@
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
+#if !defined(__sgi)
 #include <fts.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
diff -N -r -u -w fakechroot-2.20.1/src/getcwd_real.c fakechroot-2.20.1-patched/src/getcwd_real.c
--- fakechroot-2.20.1/src/getcwd_real.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/getcwd_real.c	2019-10-04 06:57:52.064613520 +0000
@@ -20,6 +20,10 @@
 
 #include <config.h>
 
+#if defined(__sgi)
+#define dirfd(dirp)((dirp)->dd_fd)
+#endif
+
 #define _BSD_SOURCE
 #define _GNU_SOURCE
 #define _DEFAULT_SOURCE
@@ -282,7 +286,11 @@
                         for (;;) {
                                 if (!(dp = readdir(dir)))
                                         goto notfound;
+#if defined(__sgi)
+                                if (dp->d_ino == ino)
+#else
                                 if (dp->d_fileno == ino)
+#endif
                                         break;
                         }
                 } else
diff -N -r -u -w fakechroot-2.20.1/src/getpeername.c fakechroot-2.20.1-patched/src/getpeername.c
--- fakechroot-2.20.1/src/getpeername.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/getpeername.c	2019-10-04 06:57:52.067483920 +0000
@@ -18,12 +18,23 @@
 */
 
 
+#if defined(__sgi)
+#include <standards.h>
+#undef _NO_XOPEN4
+#undef _NO_XOPEN5
+#undef _KERNEL
+#define _ABIAPI 1
+#include <sys/socket.h>
+#define _XPG5_GETPEERNAME_USER_DEFINED 1
+#endif
+
 #include <config.h>
 
 #ifdef HAVE_GETPEERNAME
 
 #define _GNU_SOURCE
 #include <stddef.h>
+
 #include <sys/socket.h>
 
 #ifdef AF_UNIX
@@ -42,7 +53,11 @@
 #endif
 
 
+#if defined(__sgi)
+wrapper(getpeername, int, (int s, struct sockaddr *addr, socklen_t * addrlen))
+#else
 wrapper(getpeername, int, (int s, GETPEERNAME_TYPE_ARG2(addr), socklen_t * addrlen))
+#endif
 {
     int status;
     socklen_t origlen = *addrlen;
diff -N -r -u -w fakechroot-2.20.1/src/getsockname.c fakechroot-2.20.1-patched/src/getsockname.c
--- fakechroot-2.20.1/src/getsockname.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/getsockname.c	2019-10-04 06:57:52.069745920 +0000
@@ -17,6 +17,15 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
 */
 
+#if defined(__sgi)
+#include <standards.h>
+#undef _NO_XOPEN4
+#undef _NO_XOPEN5
+#undef _KERNEL
+#define _ABIAPI 1
+#include <sys/socket.h>
+#define _XPG5_GETSOCKNAME_USER_DEFINED 1
+#endif
 
 #include <config.h>
 
@@ -42,7 +51,11 @@
 #endif
 
 
+#if defined(__sgi)
+wrapper(getsockname, int, (int s, struct sockaddr * addr, socklen_t * addrlen))
+#else
 wrapper(getsockname, int, (int s, GETSOCKNAME_TYPE_ARG2(addr), socklen_t * addrlen))
+#endif
 {
     int status;
     socklen_t origlen = *addrlen;
diff -N -r -u -w fakechroot-2.20.1/src/glob.c fakechroot-2.20.1-patched/src/glob.c
--- fakechroot-2.20.1/src/glob.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/glob.c	2019-10-04 06:57:52.072114240 +0000
@@ -17,10 +17,45 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
 */
 
+#if defined(__sgi)
+#include <standards.h>
+#include <sys/types.h>
+
+typedef struct {
+	size_t gl_pathc;	/* Count of total paths so far. */
+	char **gl_pathv;	/* List of paths matching pattern. */
+	size_t gl_offs;		/* Reserved at beginning of gl_pathv. */
+	void *gl_pad[8];
+} glob_t;
+
+#define	GLOB_APPEND	0x0001	/* Append to output from previous call. */
+#define	GLOB_DOOFFS	0x0002	/* Use gl_offs. */
+#define	GLOB_ERR	0x0004	/* Return on error. */
+#define	GLOB_MARK	0x0008	/* Append / to matching directories. */
+#define	GLOB_NOCHECK	0x0010	/* Return pattern itself if nothing matches. */
+#define	GLOB_NOSORT	0x0020	/* Don't sort. */
+#define GLOB_NOESCAPE   0x0040  /* Disable backslash escaping. */
+
+#define	GLOB_BRACE	0x0080	/* Expand braces ala csh. */
+#define	GLOB_MAGCHAR	0x0100	/* Pattern had globbing characters. */
+#define	GLOB_NOMAGIC	0x0200	/* GLOB_NOCHECK without magic chars (csh). */
+#define	GLOB_QUOTE	0x0400	/* Quote special chars with \. */
+#define	GLOB_TILDE	0x0800	/* Expand tilde names from the passwd file. */
+#define	GLOB_LIMIT	0x2000	/* Limit pattern match output to ARG_MAX */
+
+#define GLOB_NOSYS	(-1)	/* The implementation does not support this */
+#define GLOB_ABORTED    (-2)	/* The scan was stopped */
+#define	GLOB_NOSPACE	(-3)	/* Malloc call failed. */
+#define GLOB_NOMATCH    (-4)	/* No existing pathname matches */
+#define	GLOB_ABEND	(-4)	/* Unignored error. */
+
+#endif
 
 #include <config.h>
 
+#if !defined(__sgi)
 #include <glob.h>
+#endif
 #include "libfakechroot.h"
 
 
diff -N -r -u -w fakechroot-2.20.1/src/libfakechroot.c fakechroot-2.20.1-patched/src/libfakechroot.c
--- fakechroot-2.20.1/src/libfakechroot.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/libfakechroot.c	2019-10-04 06:57:52.074618160 +0000
@@ -138,6 +138,9 @@
     }
 }
 
+#if defined(__sgi)
+#define RTLD_NEXT 0
+#endif
 
 /* Lazily load function */
 LOCAL fakechroot_wrapperfn_t fakechroot_loadfunc (struct fakechroot_wrapper * w)
diff -N -r -u -w fakechroot-2.20.1/src/lstat.c fakechroot-2.20.1-patched/src/lstat.c
--- fakechroot-2.20.1/src/lstat.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/lstat.c	2019-10-04 06:57:52.077040640 +0000
@@ -28,6 +28,22 @@
 #include "lstat.h"
 
 
+#if defined(__sgi)
+wrapper(lstat, int, (const char * filename, struct stat * buf))
+{
+    debug("lstat(\"%s\", &buf)", filename);
+
+    if (!fakechroot_localdir(filename)) {
+        if (filename != NULL) {
+            char abs_filename[FAKECHROOT_PATH_MAX];
+            rel2abs(filename, abs_filename);
+            filename = abs_filename;
+        }
+    }
+
+    return lstat_rel(filename, buf);
+}
+#else
 wrapper(lstat, int, (int ver, const char * filename, struct stat * buf))
 {
     debug("lstat(%d, \"%s\", &buf)", ver, filename);
@@ -42,6 +58,7 @@
 
     return lstat_rel(ver, filename, buf);
 }
+#endif
 
 
 /* Prevent looping with realpath() */
diff -N -r -u -w fakechroot-2.20.1/src/lstat.h fakechroot-2.20.1-patched/src/lstat.h
--- fakechroot-2.20.1/src/lstat.h	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/lstat.h	2019-10-04 06:57:52.079152400 +0000
@@ -26,7 +26,11 @@
 
 #ifndef HAVE___LXSTAT
 
+#if defined(__sgi)
+wrapper_proto(lstat, int, (const char *, struct stat *));
+#else
 wrapper_proto(lstat, int, (int, const char *, struct stat *));
+#endif
 
 int lstat_rel(const char *, struct stat *);
 
diff -N -r -u -w fakechroot-2.20.1/src/mknod.c fakechroot-2.20.1-patched/src/mknod.c
--- fakechroot-2.20.1/src/mknod.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/mknod.c	2019-10-04 06:57:52.081277760 +0000
@@ -25,8 +25,14 @@
 #include <sys/stat.h>
 #include "libfakechroot.h"
 
+extern char * fakechroot_abspath;
+extern char * fakechroot_buf;
 
+#if defined(__sgi)
 wrapper(mknod, int, (const char * pathname, mode_t mode, dev_t dev))
+#else
+wrapper(mknod, int, (const char * pathname, mode_t mode, dev_t dev))
+#endif
 {
     debug("mknod(\"%s\", 0%o, %ld)", pathname, mode, dev);
     expand_chroot_path(pathname);
diff -N -r -u -w fakechroot-2.20.1/src/rel2absat.c fakechroot-2.20.1-patched/src/rel2absat.c
--- fakechroot-2.20.1/src/rel2absat.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/rel2absat.c	2019-10-04 06:57:52.083722720 +0000
@@ -55,13 +55,21 @@
 
     if (*name == '/') {
         strlcpy(resolved, name, FAKECHROOT_PATH_MAX);
-    } else if(dirfd == AT_FDCWD) {
+    }
+#if !defined(__sgi)
+    else if(dirfd == AT_FDCWD) {
         if (! getcwd(cwd, FAKECHROOT_PATH_MAX - 1)) {
             goto error;
         }
         snprintf(resolved, FAKECHROOT_PATH_MAX, "%s/%s", cwd, name);
-    } else {
+    }
+#endif
+ else {
+#if defined(__sgi)
+        if ((cwdfd = nextcall(open)(".", O_RDONLY)) == -1) {
+#else
         if ((cwdfd = nextcall(open)(".", O_RDONLY|O_DIRECTORY)) == -1) {
+#endif
             goto error;
         }
 
diff -N -r -u -w fakechroot-2.20.1/src/scandir.c fakechroot-2.20.1-patched/src/scandir.c
--- fakechroot-2.20.1/src/scandir.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/scandir.c	2019-10-04 06:57:52.085878560 +0000
@@ -26,7 +26,11 @@
 #include "libfakechroot.h"
 
 
+#if defined(__sgi)
+wrapper(scandir, int, (const char * dir, struct dirent *** namelist, int (*filter)(struct dirent *), int (*compar)(struct dirent **, struct dirent **)))
+#else
 wrapper(scandir, int, (const char * dir, struct dirent *** namelist, SCANDIR_TYPE_ARG3(filter), SCANDIR_TYPE_ARG4(compar)))
+#endif
 {
     char fakechroot_abspath[FAKECHROOT_PATH_MAX];
     char fakechroot_buf[FAKECHROOT_PATH_MAX];
diff -N -r -u -w fakechroot-2.20.1/src/scandir64.c fakechroot-2.20.1-patched/src/scandir64.c
--- fakechroot-2.20.1/src/scandir64.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/scandir64.c	2019-10-04 06:57:52.088104400 +0000
@@ -27,7 +27,11 @@
 #include "libfakechroot.h"
 
 
+#if defined(__sgi)
+wrapper(scandir64, int, (const char * dir, struct dirent64 *** namelist, int(*filter)(struct dirent64 *), int (*compar)(struct dirent64 **, struct dirent64 **)))
+#else
 wrapper(scandir64, int, (const char * dir, struct dirent64 *** namelist, SCANDIR64_TYPE_ARG3(filter), SCANDIR64_TYPE_ARG4(compar)))
+#endif
 {
     char fakechroot_abspath[FAKECHROOT_PATH_MAX];
     char fakechroot_buf[FAKECHROOT_PATH_MAX];
diff -N -r -u -w fakechroot-2.20.1/src/setenv.c fakechroot-2.20.1-patched/src/setenv.c
--- fakechroot-2.20.1/src/setenv.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/setenv.c	2019-10-04 06:57:52.090835280 +0000
@@ -43,7 +43,10 @@
    modified for uClibc by Erik Andersen <andersen@codepoet.org>
 */
 
+#if !defined(__sgi)
 #include <features.h>
+#endif
+
 #include <errno.h>
 #include <stdlib.h>
 #include <string.h>
@@ -54,6 +57,9 @@
    environment.  */
 static char **last_environ;
 
+#if defined(__sgi)
+extern char **__environ;
+#endif
 
 /* This function is used by `setenv' and `putenv'.  The difference between
    the two functions is that for the former must create a new string which
diff -N -r -u -w fakechroot-2.20.1/src/setenv.h fakechroot-2.20.1-patched/src/setenv.h
--- fakechroot-2.20.1/src/setenv.h	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/setenv.h	2019-10-04 06:57:52.092961520 +0000
@@ -24,7 +24,11 @@
 int __setenv (const char *, const char *, int);
 int __unsetenv(const char *);
 int __clearenv(void);
+#if defined(__sgi)
+int __putenv(const char *);
+#else
 int __putenv(char *);
+#endif
 
 #define setenv   __setenv
 #define unsetenv __unsetenv
diff -N -r -u -w fakechroot-2.20.1/src/stat.c fakechroot-2.20.1-patched/src/stat.c
--- fakechroot-2.20.1/src/stat.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/stat.c	2019-10-04 06:57:52.095148480 +0000
@@ -30,6 +30,8 @@
 
 #include "libfakechroot.h"
 
+extern char * fakechroot_abspath;
+extern char * fakechroot_buf;
 
 wrapper(stat, int, (const char * file_name, struct stat * buf))
 {
diff -N -r -u -w fakechroot-2.20.1/src/stat64.c fakechroot-2.20.1-patched/src/stat64.c
--- fakechroot-2.20.1/src/stat64.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/stat64.c	2019-10-04 06:57:52.097269200 +0000
@@ -31,6 +31,9 @@
 
 #include "libfakechroot.h"
 
+extern char * fakechroot_abspath;
+extern char * fakechroot_buf;
+
 
 wrapper(stat64, int, (const char * file_name, struct stat64 * buf))
 {
diff -N -r -u -w fakechroot-2.20.1/src/statfs.c fakechroot-2.20.1-patched/src/statfs.c
--- fakechroot-2.20.1/src/statfs.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/src/statfs.c	2019-10-04 06:57:52.099409360 +0000
@@ -30,14 +30,21 @@
 #endif
 #include "libfakechroot.h"
 
+extern char * fakechroot_abspath;
+extern char * fakechroot_buf;
 
+
+#if defined(__sgi)
+wrapper(statfs, int, (const char * path, struct statfs * buf, int len, int fstyp))
+#else
 wrapper(statfs, int, (const char * path, struct statfs * buf))
+#endif
 {
     char fakechroot_abspath[FAKECHROOT_PATH_MAX];
     char fakechroot_buf[FAKECHROOT_PATH_MAX];
     debug("statfs(\"%s\", &buf)", path);
     expand_chroot_path(path);
-    return nextcall(statfs)(path, buf);
+    return nextcall(statfs)(path, buf, len, fstyp);
 }
 
 #else
diff -N -r -u -w fakechroot-2.20.1/test/bin/fakechroot fakechroot-2.20.1-patched/test/bin/fakechroot
--- fakechroot-2.20.1/test/bin/fakechroot	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/test/bin/fakechroot	2019-10-04 06:57:52.101520560 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 pwd=`dirname $0`
 abs_top_srcdir=${abs_top_srcdir:-`cd "$pwd/../.." 2>/dev/null && pwd -P`}
@@ -24,5 +24,5 @@
 
 FAKECHROOT_CMD_SUBST="`echo $cmd_subst | tr ' ' ':'`${FAKECHROOT_CMD_SUBST:+:$FAKECHROOT_CMD_SUBST}"
 
-"$abs_top_srcdir/scripts/env.fakechroot" - FAKECHROOT_CMD_SUBST="$FAKECHROOT_CMD_SUBST" HOME="$HOME" PATH="$PATH" SHELL="$SHELL" "$abs_top_srcdir/scripts/fakechroot" -c "$abs_top_srcdir/scripts" -l "$abs_top_srcdir/src/.libs/libfakechroot.so" "$@"
+"$abs_top_srcdir/scripts/env.fakechroot.sh" - FAKECHROOT_CMD_SUBST="$FAKECHROOT_CMD_SUBST" HOME="$HOME" PATH="$PATH" SHELL="$SHELL" "$abs_top_srcdir/scripts/fakechroot.sh" -c "$abs_top_srcdir/scripts" -l "$abs_top_srcdir/src/.libs/libfakechroot.so" "$@"
 exit $?
diff -N -r -u -w fakechroot-2.20.1/test/chroot.sh fakechroot-2.20.1-patched/test/chroot.sh
--- fakechroot-2.20.1/test/chroot.sh	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/test/chroot.sh	2019-10-04 06:57:52.103300560 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 srcdir=${srcdir:-.}
 
diff -N -r -u -w fakechroot-2.20.1/test/fakechroot.sh fakechroot-2.20.1-patched/test/fakechroot.sh
--- fakechroot-2.20.1/test/fakechroot.sh	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/test/fakechroot.sh	2019-10-04 06:57:52.105123120 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 srcdir=${srcdir:-.}
 FAKECHROOT="${FAKECHROOT:-$srcdir/bin/fakechroot}"
diff -N -r -u -w fakechroot-2.20.1/test/src/test-fts.c fakechroot-2.20.1-patched/test/src/test-fts.c
--- fakechroot-2.20.1/test/src/test-fts.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/test/src/test-fts.c	2019-10-04 06:57:52.107351920 +0000
@@ -2,7 +2,9 @@
 #define _DEFAULT_SOURCE
 #include <sys/types.h>
 #include <sys/stat.h>
+#if !defined(__sgi)
 #include <fts.h>
+#endif
 
 #include <errno.h>
 #include <unistd.h>
@@ -11,6 +13,7 @@
 
 
 int main (int argc, char *argv[]) {
+#if !defined(__sgi)
     FTS *tree;
     FTSENT *node;
 
@@ -22,6 +25,7 @@
         exit(2);
     }
 
+
     options = atoi(argv[1]);
     paths = argv + 2;
 
@@ -44,6 +48,7 @@
         perror("fts_close");
         return 1;
     }
+#endif
 
     return 0;
 }
diff -N -r -u -w fakechroot-2.20.1/test/src/test-mkstemps.c fakechroot-2.20.1-patched/test/src/test-mkstemps.c
--- fakechroot-2.20.1/test/src/test-mkstemps.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/test/src/test-mkstemps.c	2019-10-04 06:57:52.109287360 +0000
@@ -7,6 +7,7 @@
     int suffixlen;
     int fd;
 
+#if !defined(__sgi)
     if (argc != 3) {
         fprintf(stderr, "Usage: %s template suffixlen\n", argv[0]);
         exit(2);
@@ -20,6 +21,7 @@
         exit(1);
     }
     printf("%s\n", template);
+#endif
 
     return 0;
 }
diff -N -r -u -w fakechroot-2.20.1/test/src/test-posix_spawn.c fakechroot-2.20.1-patched/test/src/test-posix_spawn.c
--- fakechroot-2.20.1/test/src/test-posix_spawn.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/test/src/test-posix_spawn.c	2019-10-04 06:57:52.111460560 +0000
@@ -2,7 +2,9 @@
 #define _DEFAULT_SOURCE
 #include <sys/types.h>
 #include <signal.h>
+#if !defined(__sgi)
 #include <spawn.h>
+#endif
 #include <string.h>
 #include <stdlib.h>
 #include <stdio.h>
@@ -16,6 +18,7 @@
         exit(2);
     }
 
+#if !defined(__sgi)
     {
         pid_t pid;
         char *newargv[] = { argv[1], argv[2], NULL };
@@ -39,4 +42,7 @@
 
     /* Execution should never reach here */
     return 1;
+#else
+    return 0;
+#endif
 }
diff -N -r -u -w fakechroot-2.20.1/test/src/test-posix_spawnp.c fakechroot-2.20.1-patched/test/src/test-posix_spawnp.c
--- fakechroot-2.20.1/test/src/test-posix_spawnp.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/test/src/test-posix_spawnp.c	2019-10-04 06:57:52.113569840 +0000
@@ -2,7 +2,9 @@
 #define _DEFAULT_SOURCE
 #include <sys/types.h>
 #include <signal.h>
+#if !defined(__sgi)
 #include <spawn.h>
+#endif
 #include <string.h>
 #include <stdlib.h>
 #include <stdio.h>
@@ -11,6 +13,7 @@
 extern char **environ;
 
 int main (int argc, char* argv[]) {
+#if !defined(__sgi)
     if (argc != 3) {
         fprintf(stderr, "Usage: %s cmd arg\n", argv[0]);
         exit(2);
@@ -39,4 +42,7 @@
 
     /* Execution should never reach here */
     return 1;
+#else
+    return 0;
+#endif
 }
diff -N -r -u -w fakechroot-2.20.1/test/src/test-statfs.c fakechroot-2.20.1-patched/test/src/test-statfs.c
--- fakechroot-2.20.1/test/src/test-statfs.c	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/test/src/test-statfs.c	2019-10-04 06:57:52.115423440 +0000
@@ -25,7 +25,11 @@
         exit(2);
     }
 
+#if defined(__sgi)
+    if (statfs(argv[1], &statfsb, sizeof(struct statfs), 0)) {
+#else
     if (statfs(argv[1], &statfsb)) {
+#endif
         perror("statfs");
         exit(1);
     }
diff -N -r -u -w fakechroot-2.20.1/test/t/00echo.t fakechroot-2.20.1-patched/test/t/00echo.t
--- fakechroot-2.20.1/test/t/00echo.t	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/test/t/00echo.t	2019-10-04 06:57:52.117304720 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 # all paths need to be relative to $srcdir variable
 srcdir=${srcdir:-.}
diff -N -r -u -w fakechroot-2.20.1/test/testtree.sh fakechroot-2.20.1-patched/test/testtree.sh
--- fakechroot-2.20.1/test/testtree.sh	2019-03-22 20:59:58.000000000 +0000
+++ fakechroot-2.20.1-patched/test/testtree.sh	2019-10-04 06:57:52.119303840 +0000
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env sh
 
 if [ $# -gt 0 ]; then
     destdir=$1
