diff -r -u -w boost_1_33_1/configure boost_1_33_1-patched/configure
--- boost_1_33_1/configure	2005-11-05 16:30:41.000000000 +0000
+++ boost_1_33_1-patched/configure	2018-12-20 14:31:59.239524400 +0000
@@ -288,8 +288,8 @@
 LIBS=$LIBS
 
 all: .dummy
-	@echo "\$(BJAM) \$(BJAM_CONFIG) -sTOOLS=\$(TOOLSET) \$(LIBS)"
-	@\$(BJAM) \$(BJAM_CONFIG) -sTOOLS=\$(TOOLSET) \$(LIBS) || \\
+	@echo "\$(BJAM) \$(BJAM_CONFIG) --layout=system -sTOOLS=\$(TOOLSET) \$(LIBS)"
+	@\$(BJAM) \$(BJAM_CONFIG) --layout=system -sTOOLS=\$(TOOLSET) \$(LIBS) || \\
 	echo "Not all Boost libraries built properly."
 
 clean: .dummy
@@ -299,11 +299,11 @@
 	rm -rf Makefile config.log
 
 check: .dummy
-	@cd status && ../\$(BJAM) \$(BJAM_CONFIG) -sTOOLS=\$(TOOLSET) test || echo "Some Boost regression tests failed. This is normal."
+	@cd status && ../\$(BJAM) \$(BJAM_CONFIG) --layout=system -sTOOLS=\$(TOOLSET) test || echo "Some Boost regression tests failed. This is normal."
 
 install: .dummy
-	@echo "\$(BJAM) \$(BJAM_CONFIG) --prefix=\$(PREFIX) --exec-prefix=\$(EPREFIX) --libdir=\$(LIBDIR) --includedir=\$(INCLUDEDIR) -sTOOLS=\$(TOOLSET) \$(LIBS) install"
-	@\$(BJAM) \$(BJAM_CONFIG) --prefix=\$(PREFIX) --exec-prefix=\$(EPREFIX) --libdir=\$(LIBDIR) --includedir=\$(INCLUDEDIR) -sTOOLS=\$(TOOLSET) \$(LIBS) install || echo "Not all Boost libraries built properly."
+	@echo "\$(BJAM) \$(BJAM_CONFIG) --prefix=\$(PREFIX) --exec-prefix=\$(EPREFIX) --libdir=\$(LIBDIR) --includedir=\$(INCLUDEDIR) --layout=system -sTOOLS=\$(TOOLSET) \$(LIBS) install"
+	@\$(BJAM) \$(BJAM_CONFIG) --prefix=\$(PREFIX) --exec-prefix=\$(EPREFIX) --libdir=\$(LIBDIR) --includedir=\$(INCLUDEDIR) --layout=system -sTOOLS=\$(TOOLSET) \$(LIBS) install || echo "Not all Boost libraries built properly."
 
 .dummy:
 
