diff -r -u -w boost_1_35_0/boost/asio/detail/socket_types.hpp boost_1_35_0-patched/boost/asio/detail/socket_types.hpp
--- boost_1_35_0/boost/asio/detail/socket_types.hpp	2008-03-12 14:12:08.000000000 +0000
+++ boost_1_35_0-patched/boost/asio/detail/socket_types.hpp	2018-12-26 12:23:36.407686840 +0000
@@ -109,6 +109,9 @@
 #  include <sys/filio.h>
 #  include <sys/sockio.h>
 # endif
+# if defined(__sgi)
+#define IOV_MAX 1024 /* FIXME: value is from sysconf IOV_MAX */
+# endif
 #endif
 #include <boost/asio/detail/pop_options.hpp>
 
diff -r -u -w boost_1_35_0/boost/lambda/detail/operator_return_type_traits.hpp boost_1_35_0-patched/boost/lambda/detail/operator_return_type_traits.hpp
--- boost_1_35_0/boost/lambda/detail/operator_return_type_traits.hpp	2006-11-29 19:21:46.000000000 +0000
+++ boost_1_35_0-patched/boost/lambda/detail/operator_return_type_traits.hpp	2018-12-26 12:23:36.465160440 +0000
@@ -859,7 +859,7 @@
 
 
 // Forward declarations are incompatible with the libstdc++ debug mode.
-#if BOOST_WORKAROUND(__GNUC__, >= 3) && defined(_GLIBCXX_DEBUG)
+#if (BOOST_WORKAROUND(__GNUC__, >= 3) && defined(_GLIBCXX_DEBUG)) || defined(__sgi)
 #include <string>
 #include <vector>
 #include <map>
diff -r -u -w boost_1_35_0/boost/test/impl/execution_monitor.ipp boost_1_35_0-patched/boost/test/impl/execution_monitor.ipp
--- boost_1_35_0/boost/test/impl/execution_monitor.ipp	2007-11-25 18:07:19.000000000 +0000
+++ boost_1_35_0-patched/boost/test/impl/execution_monitor.ipp	2018-12-26 12:23:36.424566600 +0000
@@ -51,7 +51,7 @@
 #endif
 
 // to use vsnprintf
-#if defined(__SUNPRO_CC) && defined(__SunOS)
+#if (defined(__SUNPRO_CC) && defined(__SunOS)) || defined(__sgi)
 #  include <stdio.h>
 #  include <stdarg.h>
 using std::va_list;
diff -r -u -w boost_1_35_0/boost/test/utils/fixed_mapping.hpp boost_1_35_0-patched/boost/test/utils/fixed_mapping.hpp
--- boost_1_35_0/boost/test/utils/fixed_mapping.hpp	2007-11-25 18:07:19.000000000 +0000
+++ boost_1_35_0-patched/boost/test/utils/fixed_mapping.hpp	2018-12-26 12:23:36.434944840 +0000
@@ -67,7 +67,7 @@
     typedef typename call_traits<Value>::param_type         value_param_type;
     typedef typename call_traits<Value>::const_reference    value_ref_type;
 
-#if BOOST_WORKAROUND(__DECCXX_VER, BOOST_TESTED_AT(60590042))
+#if BOOST_WORKAROUND(__DECCXX_VER, BOOST_TESTED_AT(60590042)) || defined(__sgi)
     struct p1; friend struct p1;
     struct p2; friend struct p2;
 #endif
diff -r -u -w boost_1_35_0/boost/test/utils/named_params.hpp boost_1_35_0-patched/boost/test/utils/named_params.hpp
--- boost_1_35_0/boost/test/utils/named_params.hpp	2008-03-22 21:45:55.000000000 +0000
+++ boost_1_35_0-patched/boost/test/utils/named_params.hpp	2018-12-26 12:23:36.446326760 +0000
@@ -261,7 +261,8 @@
 template<typename T, typename Source>
 inline void
 #if BOOST_WORKAROUND( __MWERKS__, BOOST_TESTED_AT( 0x3003 ) ) \
-    || BOOST_WORKAROUND( __DECCXX_VER, BOOST_TESTED_AT(60590042) )
+    || BOOST_WORKAROUND( __DECCXX_VER, BOOST_TESTED_AT(60590042) ) \
+    || defined(__sgi)
 optionally_assign( T& target, Source src )
 #else
 optionally_assign( T& target, Source const& src )
diff -r -u -w boost_1_35_0/boost/thread/pthread/thread.hpp boost_1_35_0-patched/boost/thread/pthread/thread.hpp
--- boost_1_35_0/boost/thread/pthread/thread.hpp	2008-03-21 15:04:10.000000000 +0000
+++ boost_1_35_0-patched/boost/thread/pthread/thread.hpp	2018-12-26 12:23:36.480187640 +0000
@@ -131,6 +131,22 @@
             }
         };
         
+        template<typename F>
+        struct thread_data<boost::reference_wrapper<F> >:
+            detail::thread_data_base
+        {
+            F& f;
+        
+            thread_data(boost::reference_wrapper<F> f_):
+                f(f_)
+            {}
+        
+            void run()
+            {
+                f();
+            }
+        };
+	
         mutable boost::mutex thread_info_mutex;
         detail::thread_data_ptr thread_info;
 
@@ -204,21 +220,6 @@
     }
 
 
-    template<typename F>
-    struct thread::thread_data<boost::reference_wrapper<F> >:
-        detail::thread_data_base
-    {
-        F& f;
-        
-        thread_data(boost::reference_wrapper<F> f_):
-            f(f_)
-        {}
-        
-        void run()
-        {
-            f();
-        }
-    };
 
     namespace this_thread
     {
diff -r -u -w boost_1_35_0/configure boost_1_35_0-patched/configure
--- boost_1_35_0/configure	2007-05-09 23:09:46.000000000 +0000
+++ boost_1_35_0-patched/configure	2018-12-26 12:23:36.490171720 +0000
@@ -9,9 +9,9 @@
 
 BJAM=""
 TOOLSET=""
-BJAM_CONFIG=""
+BJAM_CONFIG="${BJAM_CONFIG}"
 BUILD=""
 PREFIX=/usr/local
 EPREFIX=
 LIBDIR=
 INCLUDEDIR=
diff -r -u -w boost_1_35_0/tools/build/v2/tools/mipspro.jam boost_1_35_0-patched/tools/build/v2/tools/mipspro.jam
--- boost_1_35_0/tools/build/v2/tools/mipspro.jam	2007-11-15 16:31:06.000000000 +0000
+++ boost_1_35_0-patched/tools/build/v2/tools/mipspro.jam	2018-12-26 12:23:36.482635800 +0000
@@ -69,7 +69,7 @@
 flags mipspro.compile OPTIONS <warnings>off : -w ;
 flags mipspro.compile OPTIONS <warnings>on : -ansiW -diag_suppress 1429 ; # suppress long long is nonstandard warning
 flags mipspro.compile OPTIONS <warnings>all : -fullwarn ;
-flags mipspro.compile OPTIONS <optimization>speed : -Ofast ;
+flags mipspro.compile OPTIONS <optimization>speed : -O3 ;
 flags mipspro.compile OPTIONS <optimization>space : -O2 ;
 flags mipspro.compile OPTIONS <cflags> : -LANG:std ;
 flags mipspro.compile.c++ OPTIONS <inlining>off : -INLINE:none ;
@@ -87,7 +87,7 @@
 
 actions compile.c++
 {
-    "$(CONFIG_COMMAND)" -FE:template_in_elf_section -ptused $(OPTIONS) -D$(DEFINES) -I"$(INCLUDES)" -c -o "$(<)" "$(>)"
+    "$(CONFIG_COMMAND)" -FE:template_in_elf_section -no_prelink -ptused $(OPTIONS) -D$(DEFINES) -I"$(INCLUDES)" -c -o "$(<)" "$(>)"
 }
 
 actions compile.fortran
@@ -105,7 +105,7 @@
 flags mipspro.link OPTIONS $(cpu-arch-32) : -n32 ;
 flags mipspro.link OPTIONS $(cpu-arch-64) : -64 ;
 
-flags mipspro.link OPTIONS <optimization>speed : -Ofast ;
+flags mipspro.link OPTIONS <optimization>speed : -O3 ;
 flags mipspro.link OPTIONS <optimization>space : -O2 ;
 flags mipspro.link OPTIONS <linkflags> ;
 flags mipspro.link LINKPATH <library-path> ;
@@ -124,7 +124,7 @@
 
 actions link bind LIBRARIES
 {
-    "$(CONFIG_COMMAND)" -FE:template_in_elf_section -ptused $(OPTIONS) -L"$(LINKPATH)" -R"$(RPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -Bdynamic -l$(FINDLIBS-SA) -Bstatic -l$(FINDLIBS-ST) -B$(LINK-RUNTIME) -lm
+    "$(CONFIG_COMMAND)" -FE:template_in_elf_section -no_prelink -ptused $(OPTIONS) -L"$(LINKPATH)" -rpath"$(RPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -Bdynamic -l$(FINDLIBS-SA) -Bstatic -l$(FINDLIBS-ST) -B$(LINK-RUNTIME) -lm
 }
 
 # Slight mods for dlls
@@ -135,11 +135,11 @@
 
 actions link.dll bind LIBRARIES
 {
-    "$(CONFIG_COMMAND)" $(OPTIONS) -L"$(LINKPATH)" -R"$(RPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -Bdynamic -l$(FINDLIBS-SA) -Bstatic -l$(FINDLIBS-ST) -B$(LINK-RUNTIME)
+    "$(CONFIG_COMMAND)" $(OPTIONS) -L"$(LINKPATH)" -rpath"$(RPATH)" -shared -Wl,-set_version,sgi1.35.0 -o "$(<)" "$(>)" "$(LIBRARIES)" -Bdynamic -l$(FINDLIBS-SA) -Bstatic -l$(FINDLIBS-ST) -B$(LINK-RUNTIME)
 }
 
 # Declare action for creating static libraries
 actions piecemeal archive
 {
-    ar -cr "$(<)" "$(>)"
+    CC -ar -WR,-suv -o "$(<)" "$(>)"
 }
