diff -r -u -w boost_1_52_0/tools/build/v2/tools/mipspro.jam boost_1_52_0-patched/tools/build/v2/tools/mipspro.jam
--- boost_1_52_0/tools/build/v2/tools/mipspro.jam	2007-11-15 16:31:06.000000000 +0000
+++ boost_1_52_0-patched/tools/build/v2/tools/mipspro.jam	2018-09-15 00:35:07.448859160 +0000
@@ -124,7 +124,7 @@
 
 actions link bind LIBRARIES
 {
-    "$(CONFIG_COMMAND)" -FE:template_in_elf_section -ptused $(OPTIONS) -L"$(LINKPATH)" -R"$(RPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -Bdynamic -l$(FINDLIBS-SA) -Bstatic -l$(FINDLIBS-ST) -B$(LINK-RUNTIME) -lm
+    "$(CONFIG_COMMAND)" -FE:template_in_elf_section -ptused $(OPTIONS) -L"$(LINKPATH)" -R"$(RPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -Bdynamic -l$(FINDLIBS-SA) -shared -Bstatic -l$(FINDLIBS-ST) -B$(LINK-RUNTIME) -lm
 }
 
 # Slight mods for dlls
@@ -135,7 +135,7 @@
 
 actions link.dll bind LIBRARIES
 {
-    "$(CONFIG_COMMAND)" $(OPTIONS) -L"$(LINKPATH)" -R"$(RPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -Bdynamic -l$(FINDLIBS-SA) -Bstatic -l$(FINDLIBS-ST) -B$(LINK-RUNTIME)
+    "$(CONFIG_COMMAND)" $(OPTIONS) -L"$(LINKPATH)" -R"$(RPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -shared -Bdynamic -l$(FINDLIBS-SA) -Bstatic -l$(FINDLIBS-ST) -B$(LINK-RUNTIME)
 }
 
 # Declare action for creating static libraries
