diff -r -u -w ruby-2.5.1/addr2line.c ruby-2.5.1-patched/addr2line.c
--- ruby-2.5.1/addr2line.c	2017-12-18 16:17:17.000000000 +0000
+++ ruby-2.5.1-patched/addr2line.c	2018-08-11 11:44:35.578010240 +0000
@@ -104,6 +104,31 @@
 #define SHF_COMPRESSED 0
 #endif
 
+#include <dlfcn.h>
+#include <rld_interface.h>
+
+#ifndef _RLD_INTERFACE_DLFCN_H_DLADDR
+#define _RLD_INTERFACE_DLFCN_H_DLADDR
+typedef struct Dl_info {
+  const char * dli_fname;
+  void       * dli_fbase;
+  const char * dli_sname;
+  void       * dli_saddr;
+  int          dli_version;
+  int          dli_reserved1;
+  long         dli_reserved[4];
+} Dl_info;
+#endif
+#define _RLD_DLADDR             14
+int dladdr(void *address, Dl_info *dl);
+
+int dladdr(void *address, Dl_info *dl)
+{
+  void *v;
+  v = _rld_new_interface(_RLD_DLADDR,address,dl);
+  return (int)v;
+}
+
 int kprintf(const char *fmt, ...);
 
 typedef struct {
Only in ruby-2.5.1-patched: addr2line.c.orig
diff -r -u -w ruby-2.5.1/configure.ac ruby-2.5.1-patched/configure.ac
--- ruby-2.5.1/configure.ac	2018-01-03 18:12:16.000000000 +0000
+++ ruby-2.5.1-patched/configure.ac	2018-08-11 11:44:41.342564560 +0000
@@ -426,7 +426,7 @@
 AC_SUBST(CSRCFLAG)
 
 cc_version=
-for option in --version -v -V -qversion; do
+for option in -version --version -v -V -qversion; do
     cc_version_message=`$CC $option 2>&1`
     cc_version_status=$?
     AS_CASE($cc_version_status, [0], [:], [continue])
diff -r -u -w ruby-2.5.1/cont.c ruby-2.5.1-patched/cont.c
--- ruby-2.5.1/cont.c	2017-12-06 21:51:10.000000000 +0000
+++ ruby-2.5.1-patched/cont.c	2018-08-11 11:44:35.585262720 +0000
@@ -750,8 +750,10 @@
  */
 #if defined(MAP_STACK) && !defined(__FreeBSD__) && !defined(__FreeBSD_kernel__)
 #define FIBER_STACK_FLAGS (MAP_PRIVATE | MAP_ANON | MAP_STACK)
-#else
+#elsif !defined(__sgi)
 #define FIBER_STACK_FLAGS (MAP_PRIVATE | MAP_ANON)
+#else
+#define FIBER_STACK_FLAGS (MAP_PRIVATE)
 #endif
 
 static char*
Only in ruby-2.5.1-patched: cont.c.orig
diff -r -u -w ruby-2.5.1/io.c ruby-2.5.1-patched/io.c
--- ruby-2.5.1/io.c	2017-12-24 02:41:00.000000000 +0000
+++ ruby-2.5.1-patched/io.c	2018-08-11 11:44:35.620508400 +0000
@@ -21,6 +21,10 @@
 #include <errno.h>
 #include "ruby_atomic.h"
 
+#if defined(__sgi)
+#define IOV_MAX 1024
+#endif
+
 #undef free
 #define free(x) xfree(x)
 
Only in ruby-2.5.1-patched: io.c.orig
diff -r -u -w ruby-2.5.1/vm.c ruby-2.5.1-patched/vm.c
--- ruby-2.5.1/vm.c	2017-12-16 13:02:25.000000000 +0000
+++ ruby-2.5.1-patched/vm.c	2018-08-11 11:44:35.629835520 +0000
@@ -19,6 +19,8 @@
 #include "probes.h"
 #include "probes_helper.h"
 
+#include <dlfcn.h>
+
 VALUE rb_str_concat_literals(size_t, const VALUE*);
 
 PUREFUNC(static inline const VALUE *VM_EP_LEP(const VALUE *));
Only in ruby-2.5.1-patched: vm.c.orig
