diff -N -r -u -w nspr-4.21/nspr/configure nspr-4.21-patched/nspr/configure
--- nspr-4.21/nspr/configure	2019-03-07 10:04:05.000000000 +0000
+++ nspr-4.21-patched/nspr/configure	2019-09-26 18:59:01.125569120 +0000
@@ -872,7 +872,6 @@
 sysconfdir='${prefix}/etc'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
-
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
 docdir='${datarootdir}/doc/${PACKAGE}'
@@ -6903,7 +6902,11 @@
 
     PR_MD_CSRCS=irix.c
     PR_MD_ASFILES=os_Irix.s
+    if test "$GNU_CC"; then
+        MKSHLIB='$(LD) $(DSO_LDOPTS) -shared -soname $(notdir $@) -o $@'
+    else
     MKSHLIB='$(LD) $(DSO_LDOPTS) -rdata_shared -shared -soname $(notdir $@) -o $@'
+    fi
     STRIP="$STRIP -f"
     RESOLVE_LINK_SYMBOLS=1
     if test -n "$USE_64"; then
@@ -7292,6 +7295,11 @@
 
 	    USE_64=1
 	    ;;
+    aarch64)
+	    $as_echo "#define _ARM64_ 1" >>confdefs.h
+
+	    USE_64=1
+	    ;;
     ia64)
 	    $as_echo "#define _IA64_ 1" >>confdefs.h
 
diff -N -r -u -w nspr-4.21/nspr/configure.in nspr-4.21-patched/nspr/configure.in
--- nspr-4.21/nspr/configure.in	2019-03-07 10:04:05.000000000 +0000
+++ nspr-4.21-patched/nspr/configure.in	2019-09-26 18:59:01.135097760 +0000
@@ -1716,7 +1716,11 @@
     AC_DEFINE(HAVE_FCNTL_FILE_LOCKING)
     PR_MD_CSRCS=irix.c
     PR_MD_ASFILES=os_Irix.s
+    if test "$GNU_CC"; then
+        MKSHLIB='$(LD) $(DSO_LDOPTS) -shared -soname $(notdir $@) -o $@'
+    else
     MKSHLIB='$(LD) $(DSO_LDOPTS) -rdata_shared -shared -soname $(notdir $@) -o $@'
+    fi
     STRIP="$STRIP -f"
     RESOLVE_LINK_SYMBOLS=1
     if test -n "$USE_64"; then
diff -N -r -u -w nspr-4.21/nspr/pr/src/md/unix/irix.c nspr-4.21-patched/nspr/pr/src/md/unix/irix.c
--- nspr-4.21/nspr/pr/src/md/unix/irix.c	2019-03-07 10:04:05.000000000 +0000
+++ nspr-4.21-patched/nspr/pr/src/md/unix/irix.c	2019-09-26 18:59:01.140742800 +0000
@@ -14,6 +14,8 @@
 #include <sys/mman.h>
 #include <sys/syssgi.h>
 #include <sys/time.h>
+#undef __R4000__
+#undef R4000
 #include <sys/immu.h>
 #include <sys/utsname.h>
 #include <sys/sysmp.h>
diff -N -r -u -w nspr-4.21/nspr/pr/src/md/unix/uxrng.c nspr-4.21-patched/nspr/pr/src/md/unix/uxrng.c
--- nspr-4.21/nspr/pr/src/md/unix/uxrng.c	2019-03-07 10:04:05.000000000 +0000
+++ nspr-4.21-patched/nspr/pr/src/md/unix/uxrng.c	2019-09-26 18:59:01.142745920 +0000
@@ -131,6 +131,8 @@
 #undef PRIVATE
 #include <sys/mman.h>
 #include <sys/syssgi.h>
+#undef __R4000__
+#undef R4000
 #include <sys/immu.h>
 #include <sys/systeminfo.h>
 #include <sys/utsname.h>
diff -N -r -u -w nspr-4.21/nspr/pr/src/pthreads/ptthread.c nspr-4.21-patched/nspr/pr/src/pthreads/ptthread.c
--- nspr-4.21/nspr/pr/src/pthreads/ptthread.c	2019-03-07 10:04:05.000000000 +0000
+++ nspr-4.21-patched/nspr/pr/src/pthreads/ptthread.c	2019-09-26 19:00:27.954125240 +0000
@@ -1646,10 +1646,14 @@
     int (*dynamic_pthread_setname_np)(pthread_t, const char*);
 #endif
 
+#if defined(__sgi)
+    return PR_SUCCESS;
+#else
     *(void**)(&dynamic_pthread_setname_np) =
         dlsym(RTLD_DEFAULT, "pthread_setname_np");
     if (!dynamic_pthread_setname_np)
         return PR_SUCCESS;
+#endif
 
     /*
      * The 15-character name length limit is an experimentally determined
