diff -u -r -w binutils-2.32/config/override.m4 binutils-2.32-patched/config/override.m4
--- binutils-2.32/config/override.m4	2011-03-05 07:33:13.000000000 +0000
+++ binutils-2.32-patched/config/override.m4	2019-01-27 00:21:24.239715760 +0000
@@ -29,7 +29,7 @@
 
 dnl Ensure exactly this Autoconf version is used
 m4_ifndef([_GCC_AUTOCONF_VERSION],
-  [m4_define([_GCC_AUTOCONF_VERSION], [2.64])])
+  [m4_define([_GCC_AUTOCONF_VERSION], [2.69])])
 
 dnl Test for the exact version when AC_INIT is expanded.
 dnl This allows to update the tree in steps (for testing)
