diff -u -r -w pax-utils-1.2.4/lddtree.sh pax-utils-1.2.4-patched/lddtree.sh
--- pax-utils-1.2.4/lddtree.sh	2019-01-14 22:35:29.000000000 +0000
+++ pax-utils-1.2.4-patched/lddtree.sh	2019-11-03 21:01:56.745283960 +0000
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 # Copyright 2007-2013 Gentoo Foundation
 # Copyright 2007-2013 Mike Frysinger <vapier@gentoo.org>
 # Distributed under the terms of the GNU General Public License v2
diff -u -r -w pax-utils-1.2.4/symtree.sh pax-utils-1.2.4-patched/symtree.sh
--- pax-utils-1.2.4/symtree.sh	2019-01-14 22:35:29.000000000 +0000
+++ pax-utils-1.2.4-patched/symtree.sh	2019-11-03 21:06:27.106129040 +0000
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 argv0=${0##*/}
 
