diff -r -u -w git-2.17.0/builtin/describe.c git-2.17.0-patched/builtin/describe.c
--- git-2.17.0/builtin/describe.c	Mon Apr  2 18:44:04 2018
+++ git-2.17.0-patched/builtin/describe.c	Thu Jul  5 23:06:46 2018
@@ -470,7 +470,12 @@
 {
 	struct rev_info revs;
 	struct argv_array args = ARGV_ARRAY_INIT;
-	struct process_commit_data pcd = { null_oid, oid, dst, &revs};
+//	struct process_commit_data pcd = { null_oid, oid, dst, &revs};
+	struct process_commit_data pcd;
+	pcd.current_commit = null_oid;
+	pcd.looking_for = oid;
+	pcd.dst = dst;
+	pcd.revs = &revs;
 
 	argv_array_pushl(&args, "internal: The first arg is not parsed",
 		"--objects", "--in-commit-order", "--reverse", "HEAD",
Only in git-2.17.0-patched/builtin: describe.c~
diff -r -u -w git-2.17.0/builtin/get-tar-commit-id.c git-2.17.0-patched/builtin/get-tar-commit-id.c
--- git-2.17.0/builtin/get-tar-commit-id.c	Mon Apr  2 18:44:04 2018
+++ git-2.17.0-patched/builtin/get-tar-commit-id.c	Thu Jul  5 23:07:22 2018
@@ -3,9 +3,9 @@
  */
 #include "cache.h"
 #include "commit.h"
-#include "tar.h"
 #include "builtin.h"
 #include "quote.h"
+#include "tar.h"
 
 static const char builtin_get_tar_commit_id_usage[] =
 "git get-tar-commit-id";
Only in git-2.17.0-patched/builtin: get-tar-commit-id.c~
diff -r -u -w git-2.17.0/config.mak.uname git-2.17.0-patched/config.mak.uname
--- git-2.17.0/config.mak.uname	Mon Apr  2 18:44:04 2018
+++ git-2.17.0-patched/config.mak.uname	Thu Jul  5 23:04:17 2018
@@ -280,7 +280,7 @@
 	NO_MMAP = YesPlease
 	NO_REGEX = YesPlease
 	SNPRINTF_RETURNS_BOGUS = YesPlease
-	SHELL_PATH = /usr/gnu/bin/bash
+	SHELL_PATH = $(CONFIG_SHELL)
 	NEEDS_LIBGEN = YesPlease
 endif
 ifeq ($(uname_S),IRIX64)
@@ -298,7 +298,7 @@
 	NO_MMAP = YesPlease
 	NO_REGEX = YesPlease
 	SNPRINTF_RETURNS_BOGUS = YesPlease
-	SHELL_PATH = /usr/gnu/bin/bash
+	SHELL_PATH = $(CONFIG_SHELL)
 	NEEDS_LIBGEN = YesPlease
 endif
 ifeq ($(uname_S),HP-UX)
Only in git-2.17.0-patched: config.mak.uname.orig
diff -r -u -w git-2.17.0/run-command.c git-2.17.0-patched/run-command.c
--- git-2.17.0/run-command.c	Mon Apr  2 18:44:04 2018
+++ git-2.17.0-patched/run-command.c	Thu Jul  5 23:08:11 2018
@@ -498,7 +498,7 @@
 #endif
 }
 
-static void atfork_parent(struct atfork_state *as)
+static void gitatfork_parent(struct atfork_state *as)
 {
 #ifdef NO_PTHREADS
 	if (sigprocmask(SIG_SETMASK, &as->old, NULL))
@@ -829,7 +829,7 @@
 			child_die(CHILD_ERR_ERRNO);
 		}
 	}
-	atfork_parent(&as);
+	gitatfork_parent(&as);
 	if (cmd->pid < 0)
 		error_errno("cannot fork() for %s", cmd->argv[0]);
 	else if (cmd->clean_on_exit)
Only in git-2.17.0-patched: run-command.c~
diff -r -u -w git-2.17.0/tar.h git-2.17.0-patched/tar.h
--- git-2.17.0/tar.h	Mon Apr  2 18:44:04 2018
+++ git-2.17.0-patched/tar.h	Thu Jul  5 23:08:45 2018
@@ -1,3 +1,6 @@
+#ifndef ___TAR_H
+#define ___TAR_H
+
 #define TYPEFLAG_AUTO		'\0'
 #define TYPEFLAG_REG		'0'
 #define TYPEFLAG_LNK		'2'
@@ -23,3 +26,5 @@
 	char devminor[8];	/* 337 */
 	char prefix[155];	/* 345 */
 };
+
+#endif
