diff -u -r -w gcc-3.4.6/gcc/unwind-dw2.c gcc-3.4.6-patched/gcc/unwind-dw2.c
--- gcc-3.4.6/gcc/unwind-dw2.c	2005-01-29 11:23:09.000000000 +0000
+++ gcc-3.4.6-patched/gcc/unwind-dw2.c	2019-03-28 08:30:53.224966828 +0000
@@ -1027,7 +1027,7 @@
       /* Couldn't find frame unwind info for this function.  Try a
 	 target-specific fallback mechanism.  This will necessarily
 	 not provide a personality routine or LSDA.  */
-#ifdef MD_FALLBACK_FRAME_STATE_FOR
+#if defined(BANANA) && defined(MD_FALLBACK_FRAME_STATE_FOR)
       MD_FALLBACK_FRAME_STATE_FOR (context, fs, success);
       return _URC_END_OF_STACK;
     success:
